ios/brave-ios/Sources/BraveWallet/Extensions/RpcServiceExtensions.swift

```diff
@@ -369,44 +369,36 @@ extension BraveWalletJsonRpcService {
     for coins: [BraveWallet.CoinType],
     respectHiddenNetworksPreference: Bool = true
   ) async -> [BraveWallet.NetworkInfo] {
```

Comment:
`withTaskGroup` was causing non-deterministic order of networks list which could cause test flakiness. This will now use network order returnedfrom `GetAllNetworks` api (either core or order used in `_allNetworks` implementation in `MockJsonRpcService`), except for sorting `CoinType.sol` networks to the top (existing behaviour). Localhost is still filtered out, Bitcoin Testnet filtered out dependant on `Preferences.Wallet.isBitcoinTestnetEnabled` value as before.